### PR TITLE
Improved ICMP validation

### PR DESCRIPTION
**Pull request by zakird**
This adds validation to the ICMP validation to the RTT scan, and also extends validation to use both ICMP id and sequence number.

### CODE REVIEW

**Comment by zakird** [comment]
@dadrian @paul-pearce: tagging both of you given that this is a fairly core piece of ZMap.

**Comment by dadrian** [comment]
Were we just not setting ICMP seqno previously?

**Comment by zakird** [comment]
Nope!

File: src/probe_modules/module_icmp_echo_time.c [hunk 1]
```diff
@@ -35,184 +35,193 @@ struct icmp_payload_for_rtt {
...
+			icmp_header->icmp_code,
+			ntohs(icmp_header->icmp_cksum),
+			ntohs(icmp_header->icmp_id),
+			ntohs(icmp_header->icmp_seq));
+	fprintf_ip_header(fp, iph);
+	fprintf_eth_header(fp, ethh);
+	fprintf(fp, "------------------------------------------------------\n");
 }
 
 static int icmp_validate_packet(const struct ip *ip_hdr,
```
Comment (dadrian) [inline src/probe_modules/module_icmp_echo_time.c]: Why are there two `icmp_validate_packet` functions that seem largely identical?

Comment (paul-pearce) [inline src/probe_modules/module_icmp_echo_time.c]: This problem exists across other modules as well. In our copious amounts of spare time we should try to unify and standardize how this is done.

**Comment by paul-pearce** [comment]
The changes lgtm.

Reading this I began wondering why you switched idnum from validation[2] to validation[1]. I then began wondering why you don't use validation[0]. From digging around, it seems like there's no rhyme or reasons why we use any particular validation field across any of the probe modules. We should probably regularize how this works. e.g. starting from validation[0] we assign in the order the fields occur in the packet. 

But that's a broader issue.

**Comment by zakird** [comment]
I was confused by this too.

**Pull request merged**